modules/exporter-common/src/main/scala/io/janstenpickle/trace4cats/export/QueuedSpanCompleter.scala

```diff
@@ -27,30 +27,32 @@ object QueuedSpanCompleter {
   ): Resource[F, SpanCompleter[F]] = {
     val realBufferSize = if (bufferSize < batchSize * 5) batchSize * 5 else bufferSize
 
-    def write(inFlight: Ref[F, Int], queue: Queue[F, CompletedSpan], exporter: SpanExporter[F, Chunk]): F[Unit] =
-      queue.dequeue
-        .groupWithin(batchSize, batchTimeout)
-        .map(spans => Batch(spans))
-        .evalMap { batch =>
-          exporter.exportBatch(batch).guarantee(inFlight.update(_ - batch.spans.size))
-        }
-        .compile
-        .drain
-        .onError { case th =>
-          Logger[F].warn(th)("Failed to export spans")
-        }
-
     for {
       inFlight <- Resource.liftF(Ref.of(0))
       hasLoggedWarn <- Resource.liftF(Ref.of(false))
       queue <- Resource.liftF(Queue.bounded[F, CompletedSpan](realBufferSize))
-      _ <- Resource.make(
-        Stream
-          .retry(write(inFlight, queue, exporter), 5.seconds, _ + 1.second, Int.MaxValue)
+      _ <- Resource.make {
+        queue.dequeue
+          .groupWithin(batchSize, batchTimeout)
+          .map(spans => Batch(spans))
+          .evalMap { batch =>
+            Stream
+              .retry(
+                exporter.exportBatch(batch).onError { case th =>
+                  Logger[F].warn(th)("Failed to export spans")
+                },
+                delay = 5.seconds,
+                nextDelay = _ + 1.second,
```

Comment:
It might be a good idea to make these configurable now and possibly pick some different defaults, what do you think @CremboC?

Reply:
Yeah I was thinking exactly the same. Just wasn't sure what good defaults are, the current ones aren't ideal either I don't think .

Reply:
Yeah, maybe intital delay of 500ms, with a backoff of 100ms? To expose this further up it will need wiring through all of the exporters, so if you don't want to do that it's OK by me.

Reply:
Addressed in 605a27a